integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -83,56 +94,108 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
       .set("spark.kubernetes.driver.annotation.annotation2", "annotation2-value")
       .set("spark.kubernetes.driverEnv.ENV1", "VALUE1")
       .set("spark.kubernetes.driverEnv.ENV2", "VALUE2")
-    runSparkPiAndVerifyCompletion(driverPodChecker = (driverPod: Pod) => {
-      doBasicDriverPodCheck(driverPod)
-      assert(driverPod.getMetadata.getName === "spark-integration-spark-pi")
-
-      assert(driverPod.getMetadata.getLabels.get("label1") === "label1-value")
-      assert(driverPod.getMetadata.getLabels.get("label2") === "label2-value")
-      assert(driverPod.getMetadata.getAnnotations.get("annotation1") === "annotation1-value")
-      assert(driverPod.getMetadata.getAnnotations.get("annotation2") === "annotation2-value")
-
-      val driverContainer = driverPod.getSpec.getContainers.get(0)
-      val envVars = driverContainer
-        .getEnv
-        .asScala
-        .map { env =>
-          (env.getName, env.getValue)
-        }
-        .toMap
-      assert(envVars("ENV1") === "VALUE1")
-      assert(envVars("ENV2") === "VALUE2")
-    })
+      .set("spark.kubernetes.executor.label.label1", "label1-value")
+      .set("spark.kubernetes.executor.label.label2", "label2-value")
+      .set("spark.kubernetes.executor.annotation.annotation1", "annotation1-value")
+      .set("spark.kubernetes.executor.annotation.annotation2", "annotation2-value")
+      .set("spark.executorEnv.ENV1", "VALUE1")
+      .set("spark.executorEnv.ENV2", "VALUE2")
+
+    runSparkPiAndVerifyCompletion(
+      driverPodChecker = (driverPod: Pod) => {
+        doBasicDriverPodCheck(driverPod)
+        assert(driverPod.getMetadata.getName === "spark-integration-spark-pi")
+        checkCustomSettings(driverPod)
+      },
+      executorPodChecker = (executorPod: Pod) => {
+        doBasicExecutorPodCheck(executorPod)
+        checkCustomSettings(executorPod)
+      })
+  }
+
+  test("Run SparkPi with a test secret mounted into the driver and executor pods") {
+    createTestSecret()
+    sparkAppConf
+      .set(s"spark.kubernetes.driver.secrets.$TEST_SECRET_NAME", TEST_SECRET_MOUNT_PATH)
+      .set(s"spark.kubernetes.executor.secrets.$TEST_SECRET_NAME", TEST_SECRET_MOUNT_PATH)
+    runSparkPiAndVerifyCompletion(
+      driverPodChecker = (driverPod: Pod) => {
+        doBasicDriverPodCheck(driverPod)
+        checkTestSecret(driverPod)
+      },
+      executorPodChecker = (executorPod: Pod) => {
+        doBasicExecutorPodCheck(executorPod)
+        checkTestSecret(executorPod)
+      })
+  }
+
+  test("Run SparkPi using the remote example jar with a test secret mounted into the driver and " +
```

Comment:
Is this test using the remote example jar? I can't find it being referenced in the test itself.

Reply:
Fixed.